arbiter-engine/src/examples/token_minter.rs

```diff
@@ -0,0 +1,315 @@
+            },
+            request_to: TOKEN_ADMIN_ID.to_owned(),
+            client,
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl Strategy<Message, Message> for TokenRequester {
+    async fn sync_state(&mut self) -> Result<()> {
+        Ok(())
+    }
+
+    #[tracing::instrument(skip(self, event), fields(id = %self.id))]
+    async fn process_event(&mut self, event: Message) -> Vec<Message> {
+        trace!("Processing event for `TokenRequester` {:?}.", event);
+        if event.to == self.id {
+            if event.data == "Start" {
+                trace!("Requesting address of token: {:?}", self.token_data.name);
+                let message = Message {
+                    from: self.id.to_owned(),
+                    to: self.request_to.clone(),
+                    data: serde_json::to_string(&TokenAdminQuery::AddressOf(
+                        self.token_data.name.clone(),
+                    ))
+                    .unwrap(),
+                };
+                vec![message]
+            } else if event.data == "Mint" {
+                trace!("Requesting mint of token: {:?}", self.token_data.name);
+                let message = Message {
+                    from: self.id.to_owned(),
+                    to: self.request_to.clone(),
+                    data: serde_json::to_string(&TokenAdminQuery::MintRequest(MintRequest {
+                        token: self.token_data.name.clone(),
+                        mint_to: self.client.address(),
+                        mint_amount: 1,
+                    }))
+                    .unwrap(),
+                };
+                vec![message]
+            } else {
+                vec![]
+            }
+        } else {
+            vec![]
+        }
+    }
+}
+
+// #[async_trait::async_trait]
+// impl Strategy<Log, Message> for TokenRequester {
+//     async fn sync_state(&mut self) -> Result<()> {
+//         Ok(())
+//     }
+
+//     async fn process_event(&mut self, event: Log) -> Vec<Message> {
+//         println!("Got event: {:?}", event);
+//         let message = Message {
+//             from: "requester".to_owned(),
+//             to: self.request_to.clone(),
+//             data: serde_json::to_string(&MintRequest {
+//                 token: self.token_data.name.clone(),
+//                 mint_to: self.client.address(),
+//                 mint_amount: 1,
+//             })
+//             .unwrap(),
+//         };
+//         vec![message]
+//     }
+// }
+
+#[ignore]
+#[tokio::test(flavor = "multi_thread", worker_threads = 4)]
+async fn token_minter_simulation() {
+    // TODO: Test outline, requester requests 1, 2, 3 tokens, and the test stops
+    // when the balance is checked to be 6.
+    std::env::set_var("RUST_LOG", "trace");
+    tracing_subscriber::fmt::init();
+
+    let environment = EnvironmentBuilder::new().build();
+    let connection = Connection::from(&environment);
+    let provider = Provider::new(connection);
```

Comment:
What do you think about just passing a reference of the environment to the world constructor? Not for any particular reason other than making the sim api lean. Or maybe we could even have the environment builder return a world?

Reply:
I guess that doesn't make much sense because the clients need to be constructed around the environment, nvm.

Reply:
This is WIP. Hard to know this right way to abstract this at the moment. I'm thinking about it a lot as I work on other feats. Helpful to have other features in there so we can revisit the World set ups with actual simulation on top of them.